File: Linked_List/234-palindrome-linked-list/palindrome-linked-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    bool isPalindrome(ListNode* head) {
        string result="";
        ListNode* temp=head;
        while(temp!=NULL){
            result+=to_string(temp->val);
            temp=temp->next;
        }
        int start=0;
        int end=result.size()-1;
        int n=result.size();
        while(start<=end){
            if(result[start]==result[end]){
                start++;
                end--;
            }
            else return false;
        }
        return true;
    }
};

File: Binary_Tree/234-palindrome-linked-list/palindrome-linked-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public boolean isPalindrome(ListNode head) {

        if(head==null || head.next==null) return true;

        StringBuilder sb = new StringBuilder();

        ListNode temp = head;
        while(temp!=null){

            sb.append((char)temp.val);
            temp = temp.next;
        }

        int start=0,end=sb.length()-1;
        while(start<end){

            if(sb.charAt(start)!=sb.charAt(end)) return false;
            start++;
            end--;
        }
        return true;
        
    }
}